metrics/sinks/stackdriver/stackdriver.go

```diff
@@ -751,48 +751,50 @@ func (sink *StackdriverSink) getNodeResourceLabels(labels map[string]string) map
 	}
 }
 
-func legacyCreateTimeSeries(resourceLabels map[string]string, metadata *metricMetadata, point *sd_api.Point) *sd_api.TimeSeries {
+func legacyCreateTimeSeries(resourceLabels map[string]string, metadata *metricMetadata, point *monitoringpb.Point) *monitoringpb.TimeSeries {
 	return createTimeSeries("gke_container", resourceLabels, metadata, point)
 }
 
-func createTimeSeries(resource string, resourceLabels map[string]string, metadata *metricMetadata, point *sd_api.Point) *sd_api.TimeSeries {
-	return &sd_api.TimeSeries{
-		Metric: &sd_api.Metric{
+func createTimeSeries(resource string, resourceLabels map[string]string, metadata *metricMetadata, point *monitoringpb.Point) *monitoringpb.TimeSeries {
+	return &monitoringpb.TimeSeries{
+		Metric: &google_api5.Metric{
 			Type: metadata.Name,
 		},
 		MetricKind: metadata.MetricKind,
 		ValueType:  metadata.ValueType,
-		Resource: &sd_api.MonitoredResource{
+		Resource: &google_api4.MonitoredResource{
 			Labels: resourceLabels,
 			Type:   resource,
 		},
-		Points: []*sd_api.Point{point},
+		Points: []*monitoringpb.Point{point},
 	}
 }
 
-func (sink *StackdriverSink) doublePoint(endTime time.Time, startTime time.Time, value float64) *sd_api.Point {
-	return &sd_api.Point{
-		Interval: &sd_api.TimeInterval{
-			EndTime:   endTime.Format(time.RFC3339),
-			StartTime: startTime.Format(time.RFC3339),
+func (sink *StackdriverSink) doublePoint(endTime time.Time, startTime time.Time, value float64) *monitoringpb.Point {
+	return &monitoringpb.Point{
+		Interval: &monitoringpb.TimeInterval{
+			EndTime:   &google_protobuf2.Timestamp{Seconds: endTime.Unix(), Nanos: int32(endTime.Nanosecond())},
+			StartTime: &google_protobuf2.Timestamp{Seconds: startTime.Unix(), Nanos: int32(startTime.Nanosecond())},
 		},
-		Value: &sd_api.TypedValue{
-			DoubleValue:     &value,
-			ForceSendFields: []string{"DoubleValue"},
+		Value: &monitoringpb.TypedValue{
+			Value: &monitoringpb.TypedValue_DoubleValue{
+				DoubleValue: value,
+			},
 		},
 	}
 
 }
 
-func (sink *StackdriverSink) intPoint(endTime time.Time, startTime time.Time, value int64) *sd_api.Point {
-	return &sd_api.Point{
-		Interval: &sd_api.TimeInterval{
-			EndTime:   endTime.Format(time.RFC3339),
-			StartTime: startTime.Format(time.RFC3339),
+func (sink *StackdriverSink) intPoint(endTime time.Time, startTime time.Time, value int64) *monitoringpb.Point {
+	return &monitoringpb.Point{
+		Interval: &monitoringpb.TimeInterval{
+			EndTime:   &google_protobuf2.Timestamp{Seconds: endTime.Unix(), Nanos: int32(endTime.Nanosecond())},
+			StartTime: &google_protobuf2.Timestamp{Seconds: startTime.Unix(), Nanos: int32(startTime.Nanosecond())},
```

Comment:
Consider using [this](https://godoc.org/github.com/golang/protobuf/ptypes#TimestampProto)

Reply:
I think it actually returns different type.